src/psyclone/gocean1p0.py

```diff
@@ -595,17 +468,90 @@ def __init__(self, parent=None, topology_name="", loop_type=""):
                 suggested_name, tag, symbol_type=DataSymbol,
                 datatype=INTEGER_TYPE)
 
-        # Pre-initialise the Loop children  # TODO: See issue #440
-        self.addchild(Literal("NOT_INITIALISED", INTEGER_TYPE,
-                              parent=self))  # start
-        self.addchild(Literal("NOT_INITIALISED", INTEGER_TYPE,
-                              parent=self))  # stop
-        self.addchild(Literal("1", INTEGER_TYPE, parent=self))  # step
-        self.addchild(Schedule(parent=self))  # loop body
-
+        # Initialise bounds lookup map if it is not already
         if not GOLoop._bounds_lookup:
             GOLoop.setup_bounds()
 
+    @staticmethod
+    def create(parent, loop_type, field_name="", field_space="",
+               iteration_space="", index_offset=""):
+        '''
+        Create a new instance of a GOLoop with the expected children to
+        represent the bounds given by the loop properties.
+
+        :param parent: parent node of this GOLoop.
+        :type parent: :py:class:`psyclone.psyir.nodes.Node`
+        :param str loop_type: loop type - must be 'inner' or 'outer'.
+        :param str field_name: name of the field this loop iterates on.
+        :param str field_space: space of the field this loop iterates on.
+        :param str iteration_space: iteration space of the loop.
+
+        :returns: a new GOLoop and its children.
+        :rtype: :py:class:`psyclone.gocean1p0.GOLoop`
+        '''
+
+        # Create loop node
+        node = GOLoop(parent, loop_type, field_name, field_space,
+                      iteration_space, index_offset)
+
+        # Add start, stop, step and body and Loop children
+        node.addchild(node.lower_bound())
```

Comment:
We still need them because they change dynamically then some of the loop properties are changed (e.g. index_offset) but I added the underline prefix to make them private and removed any public use of them (which now should use the loop's start_expr /stop_expr methods).

Reply:
OK.